conf/defaults.ini

```diff
@@ -227,6 +227,13 @@ x_content_type_options = true
 # when they detect reflected cross-site scripting (XSS) attacks.
 x_xss_protection = true
 
+# Enable adding the Content-Security-Policy header to your requests.
+# CSP allows to control resources the user agent is allowed to load and helps prevent XSS attacks.
+content_security_policy = false
+
+# Set Content Security Policy template used when adding the Content-Security-Policy header to your requests.
+# $NONCE in the template includes a random nonce.
+content_security_policy_template = """script-src 'unsafe-eval' 'strict-dynamic' $NONCE;object-src 'none';font-src 'self';style-src 'self' 'unsafe-inline';img-src 'self' data:;base-uri 'self';connect-src 'self' grafana.com;manifest-src 'self';media-src 'none';block-all-mixed-content;"""
```

Comment:
* `img-src 'self' data:` looks like data URIs only, which will block google cloud images within plugin readmes
* maybe add `form-action 'self'` to prevent successful XSS hacks from working

Reply:
@torkelo What do you think of @stevenvachon's feedback?

Reply:
what does base-uri mean? and will it also block if you add data source with browser access (ie browser will directly access the data source)

Reply:
I will have to look up base-uri @torkelo, I'm not really familiar with the Content Security Policy directives. They were copied from the original PR.

Reply:
@torkelo [Here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/base-uri)'s the base-uri documentation.

The `<base>` element seems to control the base URI for all relative URIs in the document, so `base-uri 'self'` probably means that all relative URIs have to have the document origin as their base. It probably makes sense.

Reply:
> * `img-src 'self' data:` looks like data URIs only, which will block google cloud images within plugin readmes

@stevenvachon I think `img-src 'self' data:` means both URIs with same origin as the document and data URIs. Regarding Google Cloud, yeah does look that way.

Reply:
@stevenvachon I added `form-action 'self'`, thanks!